### ISSUE TITLE
feat(population): multi-satellite POD union and hierarchical POD calibration prior

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#110]
## Problem / Request
Catalogs merge detections from several instruments with different detection thresholds. The design (V.A `pod_union`, V.D "multi-satellite fusion", V.B `calibration_prior`) specifies the v1 union `P_d^∪(Q) = 1 − ∏_k (1 − P_d^k(Q))` (conditionally independent detections given `Q`) as a drop-in replacement for `P_d` in the TMTPP likelihood, a per-satellite POD calibration loader, and a hierarchical prior that injects published controlled-release calibrations `(Q₅₀, σ_Q₅₀)` per instrument. Time-of-day-binned POD is the v1 time-varying option.

## User Story
> As a user with a TROPOMI + GHGSat + Tanager catalog, I want the population fit to use the union POD with each instrument's calibrated `Q₅₀` prior, so the missing-mass correction reflects which instruments could have seen each event.

## Proposed API
```python
# population/pod.py  (extends the in-tree registry)
@dataclass(frozen=True)
class InstrumentPOD: name: str; model: str; q50: float; q50_std: float; slope: float; tod_bins: Float[Array, "24"] | None = None
def pod_union(pods: Sequence[InstrumentPOD], q, *, coverage: Bool[Array, "N K"] | None = None) -> jax.Array: ...
def calibration_prior(pods: Sequence[InstrumentPOD]) -> dict: ...   # hierarchical NumPyro priors on (q50_k, slope_k)
def load_pod_calibrations(path: Path) -> list[InstrumentPOD]: ...   # YAML/CSV of published values
```

## Implementation Steps
- [ ] Add `InstrumentPOD`, `pod_union`, `calibration_prior`, `load_pod_calibrations` in `population/pod.py`
- [ ] `tmtpp_iw_model` accepts `pods: list[InstrumentPOD]`; per-event `coverage` mask (which instruments overpassed) from the ingest module when available, else all
- [ ] Time-of-day binned POD as an optional per-instrument table
- [ ] Docstring records the union-vs-categorical-mark decision (v1 union; v2 categorical)

## Definition of Done
- [ ] Single-instrument union equals that instrument's POD exactly
- [ ] Hierarchical POD coverage: with injected calibrations, `Q₅₀` posteriors contain the truth at 95 % CI ~95 % of the time across simulated basins
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/population/test_pod_union.py::test_single_instrument_identity`
- [ ] `tests/population/test_pod_union.py::test_hierarchical_calibration_coverage` (`slow`)

## Relationships
- Parent (theme epic): #75
- Blocked by: methane_pod primitives in tree; importance-corrected TMTPP likelihood
- Related: catalog ingest (coverage mask)
